packages/sdk/src/browser-wallets/unisat/index.ts

```diff
@@ -1,24 +1,133 @@
+import { getAddressFormat } from "../../addresses";
 import { OrditSDKError } from "../../errors";
-import type { BrowserWallet } from "../types";
+import { NETWORK_TO_UNISAT_NETWORK } from "./constants";
+import type { Psbt as PsbtType } from "bitcoinjs-lib";
+import type { Network } from "../../networks/types";
+import type { BrowserWalletSignResponse, WalletAddress } from "../types";
+import type { UnisatSignPSBTOptions } from "./types";
 
+/**
+ * Checks if the browser wallet extension is installed.
+ *
+ * @returns `true` if installed, `false` otherwise.
+ */
 function isInstalled() {
   if (typeof window === "undefined") {
     throw new OrditSDKError("Cannot call this function outside a browser.");
   }
   return typeof window.unisat !== "undefined";
 }
 
-async function getAddresses() {}
+/**
+ * Gets addresses from the browser wallet.
+ *
+ * @param network Network
+ * @returns An array of WalletAddress objects.
+ */
+async function getAddresses(
+  network: Network = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new OrditSDKError("Unisat not installed.");
+  }
+
+  if (!network) {
+    throw new OrditSDKError("Invalid options provided.");
+  }
+
+  const targetNetwork = NETWORK_TO_UNISAT_NETWORK[network];
+  if (!targetNetwork) {
+    throw new OrditSDKError("Unsupported network.");
+  }
+
+  const connectedNetwork = await window.unisat.getNetwork();
+  if (connectedNetwork !== targetNetwork) {
+    await window.unisat.switchNetwork(targetNetwork);
+  }
+
+  const accounts = await window.unisat.requestAccounts();
+  const publicKey = await window.unisat.getPublicKey();
+
+  const address = accounts[0];
+  if (!address) {
+    return [];
+  }
+  const format = getAddressFormat(address, network);
+  return [
+    {
+      pub: publicKey,
+      address,
+      format,
+    },
+  ];
+}
+
+/**
+ * Signs a Partially Signed Bitcoin Transaction (PSBT).
+ * To learn more, visit https://github.com/bitcoin/bitcoin/blob/master/doc/psbt.md
+ *
+ * @param psbt Partially Signed Bitcoin Transaction
+ * @param options Options for signing
+ * @returns An object containing `base64` and `hex` if the transaction is not extracted, or `hex` if the transaction is extracted.
+ */
+async function signPsbt(
+  psbt: PsbtType,
+  options: UnisatSignPSBTOptions = {},
```

Comment:
Yep I agree on Nishant's comment here. Clearer as well.

Reply:
I named it to `options` so that it matches the code comments above. I was thinking that this signature pattern for this function would be used across all browser-wallets, which would make the API consistent.